runtime/doc/ui.txt

```diff
@@ -45,13 +45,18 @@ Nvim sends msgpack-rpc notifications to all attached UIs, with method name
 Each update event is itself an array whose first element is the event name and
 remaining elements are event-parameter tuples. This allows multiple events of
 the same kind to be sent in a row without the event name being repeated. This
-batching is mostly used for "put", because each "put" event puts contents in
-one screen cell, but clients must be prepared for multiple argument sets being
-batched for all event kinds.
-
-Events must be handled in-order. The user should only see the updated screen
-state after all events in the same "redraw" batch are processed (not any
-intermediate state after processing only part of the array).
+batching is mostly used for "grid_line", because each "grid_line" event puts
+contents in one screen line, but clients must be prepared for multiple argument
+sets being batched for all event kinds.
+
+Events must be handled in-order. A "flush" event is sent when nvim is done
+redrawing the entire screen (so that all windows has a consistent view of
+buffer state, options etc). Currently every "redraw" batch ends with a single
```

Comment:
We really should avoid using things like "currently" in the docs. It doesn't help, and if it stays that way for years then it's just an awkward TODO.